src/ReactiveUI.XamForms/ShellViewModel.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) 2021 .NET Foundation and Contributors. All rights reserved.
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using Splat;
+using Xamarin.Forms;
+
+namespace ReactiveUI.XamForms
+{
+    /// <summary>
+    /// ShellViewModel.
+    /// </summary>
+    /// <typeparam name="TViewModel">The type of the view model.</typeparam>
+    /// <seealso cref="ShellContent" />
+    public class ShellViewModel<TViewModel> : ShellContent, IActivatableView
+        where TViewModel : class, new()
+    {
+        /// <summary>
+        /// The contract property.
+        /// </summary>
+        public static readonly BindableProperty ContractProperty = BindableProperty.Create(
+            nameof(Contract),
+            typeof(string),
+            typeof(ShellViewModel<TViewModel>),
+            null);
+
+        /// <summary>
+        /// The view model property.
+        /// </summary>
+        public static readonly BindableProperty ViewModelProperty = BindableProperty.Create(
+            nameof(ViewModel),
+            typeof(TViewModel),
+            typeof(ShellViewModel<TViewModel>),
+            default(TViewModel),
+            BindingMode.Default,
+            propertyChanged: ViewModelChanged);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ShellViewModel{TViewModel}"/> class.
+        /// </summary>
+        public ShellViewModel()
+        {
+            ViewLocator = Locator.Current.GetService<IViewLocator>();
```

Comment:
Setting this using the `Locator` would cause issues for anyone consuming this, not using the `Locator`. I believe `ViewLocator.Current` is the way we are handling this other places in the code.

Reply:
I Believe that I require the Locator to resolve the IViewFor<TViewModel>, you were right that my first version was not ideal, it would not have been able to deal with interfaces as the type of TViewModel. The new version should allow this but does rely upon the Locator.